### PR TITLE
feature(CombiningBuilder): add optional `includePartName` param

### PR DESCRIPTION
**Pull request by kevmoo**
Includes the name of the source parts in a comment before its content
Nice when debugging

### CODE REVIEW

**Comment by natebosch** [comment]
When do you think we'd want to avoid this? Should we just always have it turned on?

**Comment by kevmoo** [comment]
I think it'd just add noise in the default case – duplicate content from the child builder.

File: source_gen/lib/builder.dart [hunk 1]
```diff
@@ -38,12 +38,20 @@ Builder combiningBuilder([BuilderOptions options]) {
...
         '.dart': const [_outputExtensions]
       };
 
-  const CombiningBuilder();
+  /// Returns a new [CombiningBuilder].
+  ///
+  /// If [includePartName] is `true`, the name of each source part file
+  /// is output as a comment before its content. This can be useful when
+  /// debugging build issues.
+  const CombiningBuilder({bool includePartName = false})
```
Comment (natebosch) [inline source_gen/lib/builder.dart]: how would we use this? It's not plumbed through the options

File: source_gen/lib/builder.dart [hunk 2]
```diff
@@ -78,8 +86,13 @@ class CombiningBuilder implements Builder {
     });
 
     var assets = await new Stream.fromIterable(assetIds)
-        .asyncMap(buildStep.readAsString)
-        .map((s) => s.trim())
+        .asyncMap((id) async {
+          var content = (await buildStep.readAsString(id)).trim();
+          if (_includePartName) {
+            content = '// Part: ${id.pathSegments.last}\n$content';
```
Comment (natebosch) [inline source_gen/lib/builder.dart]: do we want to omit empty parts?

Comment (kevmoo) [reply source_gen/lib/builder.dart]: Nope. This is intentional. Helps folks see that there was a part – just the content was empty.

File: source_gen/lib/builder.dart (hunk 1 above)
Comment (kevmoo) [reply source_gen/lib/builder.dart]: Oops...ack...

Comment (kevmoo) [reply source_gen/lib/builder.dart]: PTAL

**Pull request merged**